### DISCUSSION TITLE
DepotDownloader.exe not doing anything?

### DISCUSSION THREAD

**Discussion by Sin-Shadow-Fox** [SteamRE/DepotDownloader#512] [Q&A]
When i click on DepotDownloader.exe, command prompt shows up for a second before closing.

**Comment by voided** [comment]
See https://github.com/SteamRE/DepotDownloader/discussions/440#discussioncomment-6246671

**Reply by Sin-Shadow-Fox** [reply]
> See [#440 (comment)](https://github.com/SteamRE/DepotDownloader/discussions/440#discussioncomment-6246671)

dotnet DepotDownloader.dll -app 754890 -depot 754891 -manifest 3746330258963937358  -username ???????? -password ???????????
>>> command not found: dotnet

**Comment by Sin-Shadow-Fox** [comment]
So i got confirmation from @teo-tsirpanis that DepotDownloader.exe works on my computer so why isn't Steam console working when i put in the command?

**Reply by Sin-Shadow-Fox** [reply]
@voided

**Reply by voided** [reply]
@Sin-Shadow-Fox DepotDownloader is just a normal command line tool. You shouldn't be trying to execute it through the Steam console. You'd just use the windows command prompt or powershell.

**Reply by Sin-Shadow-Fox** [reply]
@voided What? Then why was i taught how to enable Steam Console? How does DepotDownloader install stuff to my steam client?

**Reply by Sin-Shadow-Fox** [reply]
@voided I tried putting the command in the normal command prompt even though it shouldn't work and:

The command could not be loaded, possibly because:
     * You intended to execute a .NET application:
	The application 'DepotDownloader.dll' does not exist.
     * You intended to execute a .NET SDK command:
	No .NET SDKs were found.

Download a .NET SDK:
https://aka.ms/dotnet/download

Learn about SDK resolution:
https://aka.ms/dotnet/sdk-not-found

**Reply by Sin-Shadow-Fox** [reply]
@voided Hello? Are my @s not working?

**Comment by Luckz** [comment]
You wrote elsewhere on Github that you installed various runtimes in your Windows 7. There you also yourself linked https://steamcommunity.com/sharedfiles/filedetails/?id=2353930763 as a guide specifically for the Depot Downloader that explains in Step 6 how to launch a command prompt where you would then run the "dotnet" (and so on) command. You can right-click to paste from the clipboard into a command prompt.

However, the command you pasted above `dotnet DepotDownloader.dll -app 754890 -depot 754891 -manifest 3746330258963937358` indicates that you just want to get the very newest update for Firmament. Why?

----

It would be easier for you to just go to the SteamDB page for Firmament's core depot https://steamdb.info/depot/754891/manifests/ , to click "Steam console" as format, and to press the copy button on the right of whatever version you actually want. That gets you something like:
download_depot 754890 754891 24782735360831303
which you can just paste into the Steam Console that you already know how to access.

**Reply by Sin-Shadow-Fox** [reply]
@Luckz Hi! Glad to see some activity here. I'll try to answer your questions to the best of my ability.

First off, concerning the guide, it didn't turn out to be very helpful and someone even told me it's out of date.
Second off, The reason i want to play Firmament is because I'm a massive fan of the Myst series. I know it's their weakest title but i want to play it anyway.
And lastly, i put your command into steam console and got this

] download_depot 754890 754891 24782735360831303
Depot download failed : missing license for depots 754891

**Reply by Sin-Shadow-Fox** [reply]
@Luckz on Oct 14
> So you don't own Firmament. Steam is not a piracy tool.

![Capture](https://github.com/user-attachments/assets/123b1999-c2a1-4c7c-aaf7-5e1e7a6de3fa)

Maybe don't assume someone is a thief. It doesn't make you look good.

**Reply by Luckz** [reply]
The one for the newest version is
`download_depot 754890 754891 3746330258963937358`
But again, if it says you don't own the depot 754891 ...

What is the situation of this Win7 Steam of yours, can it not download games or what's going on here?

Did you try opening a commandline window in the folder where you put DepotDownloader?

**Reply by Sin-Shadow-Fox** [reply]
> The one for the newest version is `download_depot 754890 754891 3746330258963937358`

Same result

> What is the situation of this Win7 Steam of yours, can it not download games or what's going on here?

If you're asking why i don't just download the game from the steam client's store, it's because Steam tried and failed to kick a bunch of windows 7 users off the platform but we all took our clients into offline mode (initiating all of our games first so that they'll work in offline mode) before the "cutoff" date, so normal downloads won't work.

> Did you try opening a commandline window in the folder where you put DepotDownloader?

I'm not sure what this means. This is what i did.

![Capture](https://github.com/user-attachments/assets/71f83fee-faf0-4ed5-bb4f-7d0b3844aa1b)

**Reply by Luckz** [reply]
If your Steam is both severely outdated and not connected to the internet due to offline mode, it's presumably unlikely for it to be able to do those depot downloads.

So let's go back to https://steamcommunity.com/sharedfiles/filedetails/?id=2353930763
- as in Step 2 download https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.3/DepotDownloader-windows-x64.zip (you are using 64-bit Windows 7 and not 32-bit Windows 7, right..?)
- unpack this to a folder in File Explorer and navigate to that folder so that you see the DepotDownloader.exe and LICENSE files.
- in that folder, type `cmd.exe` into the address bar to open a command prompt, or shift-rightclick into the white empty area of the folder and choose to open a command prompt or powershell prompt
- paste into there with rightclick: `DepotDownloader.exe -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????` with your Steam login and password replacing the question marks

If that does not work due to your Windows being too old, you can try the same with an older Depot Downloader:
- download and unzip https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.5.0/DepotDownloader-windows-x64.zip
- (open a command prompt as above)
- (paste the same command)

If you manage to download it, at the very least you need to create a text file `steam_appid.txt` containing the text `754890`. Hopefully clicking on the game exe will launch Firmament then, who knows. It's quite possible though that your Steam will not know you own the game because you keep it airgapped, and thus will not let you play it either. 🤷🏼‍♂️

----

Is your hardware like 12-18 years old or something, or why do you insist on using Windows 7 in this day and age? Using Windows 10 or some Linux would require far fewer crutches.

**Reply by Sin-Shadow-Fox** [reply]
>     * as in Step 2 download https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.3/DepotDownloader-windows-x64.zip (you are using 64-bit Windows 7 and not 32-bit Windows 7, right..?)

Windows 7 Home Premium 6.1 (Build 7601: Service Pack 1) x64

>     * unpack this to a folder in File Explorer and navigate to that folder so that you see the DepotDownloader.exe and LICENSE files.

I didn't notice any file in that folder named LICENSE.

>     * in that folder, type `cmd.exe` into the address bar to open a command prompt, or shift-rightclick into the white empty area of the folder and choose to open a command prompt or powershell prompt
> 
>     * paste into there with rightclick: `DepotDownloader.exe -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????` with your Steam login and password replacing the question marks

![Capture](https://github.com/user-attachments/assets/71fb422d-2de7-4371-ba6d-d62bafd0e569)

It's been a few months since i messed with .NET. Do i have an SDK?

> If that does not work due to your Windows being too old, you can try the same with an older Depot Downloader:
> 
>     * download and unzip https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.5.0/DepotDownloader-windows-x64.zip

Why do you believe that 2.6.0 (the one i have) won't work but 2.5.0 will?

> Is your hardware like 12-18 years old or something, or why do you insist on using Windows 7 in this day and age? Using Windows 10 or some Linux would require far fewer crutches.

Aside from my motherboard and processor, i keep my parts fairly well updated. As for why i use Win7? Simply put, it's the best available option right now. It's far more stable and secure than Windows 10/11 and simpler to use than Linux. I would move to Linux if it weren't for the complexety and skill floor required to use it and the "wild west" nature of the . . .  "Distros"? Also i don't know what you mean by crutches, could you elaborate?

**Reply by Luckz** [reply]
On your screenshot, you aren't using the command I wrote (DepotDownloader.exe ...) but a different one (dotnet DeepotDownloader.dll).

> Why do you believe that 2.6.0 (the one i have) won't work but 2.5.0 will?

I linked 2.7.3 (newest) and 2.5.0 (before NET 8.0). Please try the ones I linked zips for with `DepotDownloader.exe -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????`.

I guess you can also try the command you have been attempting with https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.3/DepotDownloader-framework.zip

----

> It's far more stable and secure than Windows 10/11 and simpler to use than Linux.

It's really not any of these, 
Windows 10 is alright to use after someone hits it with a big sledgehammer. Any random "debloater" script, while likely to make a bit of a mess of the OS, is sure to satisfy your desires for having *evil Microsoft spyware* shut up. 

Are you at least using https://github.com/i486/VxKex ? Might even get to run regular Steam: https://www.reddit.com/r/windows7/comments/1fxof6b/newest_beta_steam_running_under_windows_7/
Their discord server is `CZfDcF7HQW`.

**Reply by Sin-Shadow-Fox** [reply]
> On your screenshot, you aren't using the command I wrote (DepotDownloader.exe ...) but a different one (dotnet DeepotDownloader.dll).

I . . . have no idea how i fucked that up. My best guess is when i went to copy/paste your command i must have scrolled up too high and copy/pasted my own command without thinking about it . . . Sorry.

![Steam Luckz test 1](https://github.com/user-attachments/assets/00be9264-7b19-4d38-9f55-87a1de730e58)

> > Why do you believe that 2.6.0 (the one i have) won't work but 2.5.0 will?
> 
> I linked 2.7.3 (newest) and 2.5.0 (before NET 8.0). Please try the ones I linked zips for with `DepotDownloader.exe -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????`.

Do i need to uninstall 2.6.0 first?

> > It's far more stable and secure than Windows 10/11 and simpler to use than Linux.
> 
> It's really not any of these,

I've done extensive testing and worked with people who used Win10/11 (Some of which have now switched to Win7 or Linux) and it really is. Not only does Win7 crash far less often, if at all, but I've seen a Linux distro setup (the nightmares still haunt me). Ya wanna know something really strange though? Win7 actually became MORE stable AFTER Microsoft dropped support for it . . . I'm still trying to figure that one out.

> Windows 10 is alright to use after someone hits it with a big sledgehammer. Any random "debloater" script, while likely to make a bit of a mess of the OS, is sure to satisfy your desires for having _evil Microsoft spyware_ shut up.

XD "hits it with a big sledgehammer". "having _evil Microsoft spyware_ shut up". Oh Gods this made my day, thank you lol.
But yeah, I've seen Mudahar (SomeRandomGamer) debloat Win11 so i know that's possible but there are MANY reasons why Win7 is better. Like i've said, I've done my own testing and research on the matter

> Are you at least using https://github.com/i486/VxKex ? Might even get to run regular Steam: https://www.reddit.com/r/windows7/comments/1fxof6b/newest_beta_steam_running_under_windows_7/ Their discord server is `CZfDcF7HQW`.

\*raises eyebrow\* . . . I'm listening.

**Reply by Luckz** [reply]
Important: edit your post to remove your image.
Try with the 2.7.3 zip since I assume you are using the 2.6.0 one there?

Just unzip the Depot Downloader into different folders per version to be less confused by it, and make sure you open your command prompt in the correct one.

----

For gaming at least, W10 has a boatload of things W7 just doesn't have and can't do. Whether that's DX12 stuff or the convenience of tabbing between games and other software.
Windows 11 has some new features as well, but they're less must-have (and [perhaps buggy](https://www.reddit.com/r/nvidia/comments/18vm6pm/is_it_still_recommended_to_disable_mpo_multiplane/)).
(There's also Linux distros intended for gaming, but as you said, picking one is a science in itself.)

For Windows 7 stuff, have a look around on that Discord, or just try VxKex by itself. I can't help you with that.

**Reply by Sin-Shadow-Fox** [reply]
> Important: edit your post to remove your image.

Whoops. Forgot to edit it this time. Thanks for catching that ^_^;,,

> Try with the 2.7.3 zip since I assume you are using the 2.6.0 one there?

Yeah, that was 2.6.0. This is what happened with 2.7.3. At least there is a LICENSE file there now.
![Steam Luckz test 2](https://github.com/user-attachments/assets/19c00365-0602-4efd-897f-e53835406b94)


> For gaming at least, W10 has a boatload of things W7 just doesn't have and can't do. Whether that's DX12 stuff or the convenience of tabbing between games and other software. Windows 11 has some new features as well, but they're less must-have (and [perhaps buggy](https://www.reddit.com/r/nvidia/comments/18vm6pm/is_it_still_recommended_to_disable_mpo_multiplane/)). (There's also Linux distros intended for gaming, but as you said, picking one is a science in itself.)

I've actually gotten games that say they "need" Win10/11 to work to work on Win7. And from what i hear DX12 is a mess regardless (or was it DX13?)
I'm not sure i would qualify tabbing as a "must have" either. I mean alt tabbing works just fine.

> For Windows 7 stuff, have a look around on that Discord, or just try VxKex by itself. I can't help you with that.

Fair enough. At least i now know it exists so thank you for that ^_^

**Reply by Luckz** [reply]
And https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.3/DepotDownloader-framework.zip ran like `dotnet DepotDownloader.dll -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????`? Or does your AV dislike that too?

Btw https://www.virustotal.com/gui/file/cb5be9e5a67067c499b26e6f80ad5934fb670ccf1191a10bfa5a8b66a71b2059 's Avast doesn't even hate it.

**Reply by Sin-Shadow-Fox** [reply]
> And https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.7.3/DepotDownloader-framework.zip ran like `dotnet DepotDownloader.dll -app 754890 -depot 754891 -manifest 3746330258963937358 -username ???????? -password ???????????`? Or does your AV dislike that too?

Before moving onto the framework version, i'd like to go over what FileRepMalware [Misc] is as i need to know whether to report it as a false positive or scan my PC (that window is still open)

> Btw https://www.virustotal.com/gui/file/cb5be9e5a67067c499b26e6f80ad5934fb670ccf1191a10bfa5a8b66a71b2059 's Avast doesn't even hate it.

Zillya
Trojan.Rozena.Win32.230500

Avast
FileRepMalware [Misc]

AVG
FileRepMalware [Misc]

Bkav Pro
W64.AIDetectMalware

SecureAge
Malicious

**Reply by Luckz** [reply]
Why would you scan your PC? You did not even run it.

This seems to happen for all sorts of software like VLC https://www.reddit.com/r/antivirus/comments/1bsyw00/virus_total_virus_in_vlc_zillya/ and Notepad++ https://community.notepad-plus-plus.org/topic/23697/trojan-rozena-win32-164323_npp-8-4-6-installer-x64-exe
Personally I do not pay attention to this at all if it's not at least 5–10 detections. 

For example, this is a game anti cheat: https://www.virustotal.com/gui/file/93930bcdc59b62dd0a2901f4ab88cb2bc90372b6634d237ebeb2f9fea0852b34 (2 detections)
This is a program that reads text files in real time https://www.virustotal.com/gui/file/7d1c1775f84d6f348ff3a044a06d0ba4101542132b21a6b7d2ff6b93e38f1121 (2 detections)
This is a program that searches through text files https://www.virustotal.com/gui/file/30b7f5fb2e03d8cc2f8b94eba5ad2afed317cf3e17d07c8841b858bd9d32636c (2 detections)

But if you are not comfortable with it due to the message from your AV, try the framework one. Or don't. Up to you.

`FileRepMalware` specifically just seems to mean that the antivirus does not know the particular program.

**Reply by Sin-Shadow-Fox** [reply]
> Why would you scan your PC? You did not even run it.

I think there has been a misunderstanding. It didn't get flagged when i downloaded 2.7.3. It got flagged when i ran it in cmd.exe. Also all the files in the folder i unzipped it to disappeared except the LICENSE file that appeared when i ran it.

> `FileRepMalware` specifically just seems to mean that the antivirus does not know the particular program.

So i should report as a false positive?

**Reply by Luckz** [reply]
IMO it's safe and thus a false positive. To reinforce the impression I offered examples of other commonly used programs that can trigger detections.

**Reply by Sin-Shadow-Fox** [reply]
> IMO it's safe and thus a false positive. To reinforce the impression I offered examples of other commonly used programs that can trigger detections.

Alright. I've reported it as a false positive and my client has sent off the file for review. Hopefully it'll be returned with no issues.

**Reply by Luckz** [reply]
You're not planning to wait however long it takes until they do some reviewing for you, are you?
(Did you update the definitions and all that?)

**Reply by Sin-Shadow-Fox** [reply]
> You're not planning to wait however long it takes until they do some reviewing for you, are you? 

Well yeah. If i want the file back.

> (Did you update the definitions and all that?)

Definitions?

**Comment by Sin-Shadow-Fox** [comment]
@Luckz I understand if your busy right now but you've left me in a rather precarious position. I can't turn off or restart my computer until i do something with this Avast prompt.
